### ISSUE TITLE
Bug: [strict-boolean-expressions] handle mixed enums under `allowNullableEnum` options

### ISSUE THREAD

**Issue by kozlovvski** [typescript-eslint/typescript-eslint#6737]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.9.3&sourceType=module&code=KYOwrgtgBAogHgQwgBwDbBuCBZAlnYAEygG8AoKKAFQAtcBnKAXigEYAaC6mhAF2agByAEyDOAXzJkAxgHsQ9fgHdcvGpkgCAFADcEqMMAD8ALliIU6DTnxEAlMwB8pLrgBmUXfsMOSUSeJAA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6ZfKsugIwHs+SAIZN6AD2LQUySnyaoMAbUgB3IdCaQANJihD48PioByCeEJ5IAok1gBbdGE6xEYAL4BdcO5BugA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
enum ExampleEnumMixed {
  This = 1,
  That = '2',
}

const withEnum = (value?: ExampleEnumMixed) => {
  if (value) { }
}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules": {
    "@typescript-eslint/strict-boolean-expressions": ["warn", { "allowNullableEnum": true }]
  }
};
```

### Expected Result

No warning for the if statement

### Actual Result

Warning from strict-boolean-expressions rule with { allowNullableEnum: true } option appears.

### Additional Info

Similarly to what has been done in https://github.com/typescript-eslint/typescript-eslint/pull/6618, we should also consider mixed enums in `allowNullableEnum` option.

**Comment by armano2** [comment]
this issue seem to appear only for mixed enums, string + number
